### PR TITLE
MOB-1608 - Load communities list

### PR DESCRIPTION
**Pull request by Oleg-Pecheneg**

### CODE REVIEW

File: unstoppable-ios-app/domains-manager-ios/Services/MessagingService/SubServices/MessagingAPI/PushMessagingAPIService.swift [hunk 1]
```diff
@@ -85,14 +85,61 @@ extension PushMessagingAPIService: MessagingAPIServiceProtocol {
...
-        let pushChats = try await getPushChatsForUser(user,
-                                                      page: page,
-                                                      limit: limit,
-                                                      isRequests: false)
-        
-        return try await transformPushChatsToChats(pushChats,
-                                                   isApproved: true,
-                                                   for: user)
+        try await getCommunitiesListForUser(user)
+//        let pushChats = try await getPushChatsForUser(user,
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Services/MessagingService/SubServices/MessagingAPI/PushMessagingAPIService.swift:89]: Dead?

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Entities/Chats/MessagingChatDisplayInfo.swift [hunk 2]
```diff
@@ -59,4 +60,20 @@ extension Array where Element == MessagingChatDisplayInfo {
         filter { $0.isApproved }
     }
     
+    func splitCommunitiesAndOthers() -> (chats: [Element], communities: [Element]) {
+        var chats: [Element] = []
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Entities/Chats/MessagingChatDisplayInfo.swift:64]: May be faster:

```
var runningResult: (chats: [Element], communities: [Element]) = ([], [])
return self.reduce(into: runningResult) { result, element in
switch element.type {
             case .community:
                 result.communities.append(element)
             default:
                 result.chats.append(element)
             }
}
```

**Review by rommex — APPROVED** [review]
With comments

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/ChatsList/Cells/ChatListEmptyCell/ChatListEmptyCell.swift [hunk 3]
```diff
@@ -73,6 +87,8 @@ private extension ChatListEmptyCell {
             switch dataType {
             case .chats:
                 return String.Constants.messagingChatsListEmptyTitle.localized()
+            case .communities:// TODO: - Communities
```
[the remark this answers is not in the document]
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Modules/Messaging/ChatsList/Cells/ChatListEmptyCell/ChatListEmptyCell.swift:90]: This is a reference for future updates that can't be done within this task. All of TODOs should be handled before merging into **development**

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/ChatsList/ChatsListViewController.swift [hunk 4]
```diff
@@ -524,12 +533,14 @@ extension ChatsListViewController {
...
     enum DataType: String, Hashable {
-        case chats, channels
+        case chats, communities, channels
         
         var title: String {
             switch self {
             case .chats:
                 return String.Constants.chats.localized()
+            case .communities:
+                return "Communities"
```
[the remark this answers is not in the document]
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Modules/Messaging/ChatsList/ChatsListViewController.swift:543]: Will extract when design will be ready and copy will be confirmed. Will add TODO to not forget.

File: unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Entities/Chats/MessagingChatDisplayInfo.swift (hunk 2 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Modules/Messaging/Entities/Chats/MessagingChatDisplayInfo.swift]: Updated

File: unstoppable-ios-app/domains-manager-ios/Services/MessagingService/SubServices/MessagingAPI/PushMessagingAPIService.swift (hunk 1 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Services/MessagingService/SubServices/MessagingAPI/PushMessagingAPIService.swift:89]: Updated

**Pull request merged**